File: entities/attacker.js
class Attacker extends Mob {
    get _movementCost() {
        return this.kind._movementCost;
    }

    get _energyFromPrey() {
        return this.kind._energyFromPrey;
    }

    move(pos) {
        this._gameLogic.swap(this, pos)
        this.energy -= this._movementCost;
    }

    eat(pos) {
        this.kill(pos);
        this._gameLogic.replace(pos, Air);
        this._gameLogic.swap(this, pos);
        this.energy += this._energyFromPrey;
    }
}

File: entities/attacker.py
from abc import ABCMeta, abstractmethod

from entities.mob import Mob


class Attacker(Mob, metaclass=ABCMeta):
    __slots__ = ()

    @property
    @abstractmethod
    def _movement_cost(self):
        pass

    @property
    @abstractmethod
    def _energy_from_prey(self):
        pass

    def _move(self, pos):
        self.field.swap(self.pos, pos)
        self.energy -= self._movement_cost

    def _eat(self, pos):
        self._kill(pos)
        self.field.swap(self.pos, pos)
        self.energy += self._energy_from_prey
